plugins/de.fraunhofer.ipa.rossystem.deployment/src/de/fraunhofer/ipa/rossystem/deployment/RosInstallCompiler.xtend

```diff
@@ -22,10 +22,10 @@ class RosInstallCompiler {
 	
  def compile_toRosInstall (RosSystem system,ComponentStack stack) '''«generator_helper.init_pkg()»
 «IF stack===null»«FOR repo:system.listOfRepos»
-- git: {local-name: «repo.substring(repo.lastIndexOf("/") + 1).replace(".git","")», uri: '«repo»'}
+- git: {local-name: «repo.substring(repo.lastIndexOf("/") + 1).replace(".git","")», uri: '«repo»'«IF repo.indexOf(":", repo.indexOf(":") + 1) > 1», version: '«repo.substring(repo.lastIndexOf(':') + 1)»'«ENDIF»}
```

Comment:
I think you have also to modify the uri value. You have to remove the substring ":banch_name"

if the value FromGitRepo is "https://github.com/UniversalRobots/Universal_Robots_ROS_Driver:WhatEver"

the url has to be https://github.com/UniversalRobots/Universal_Robots_ROS_Driver and the version WhatEver.

Reply:
Good point! I updated